kernel/page_table_entry/src/lib.rs

```diff
@@ -93,15 +114,47 @@ impl PageTableEntry {
     ///
     /// Note: this performs no checks about the current value of this page table entry.
     pub fn set_entry(&mut self, frame: AllocatedFrame, flags: PteFlagsArch) {
-        self.0 = (frame.start_address().value() as u64) | flags.bits();
+        #[cfg(target_arch = "x86_64")]
+        {
+            self.0 = (frame.start_address().value() as u64) | flags.bits();
+        }
+
+        #[cfg(target_arch = "aarch64")]
+        {
+            let mut flags = flags;
+
+            // telling the MMU that this is not a Block Descriptor
+            flags.insert(PteFlagsArch::PAGE_DESCRIPTOR);
```

Comment:
ok, now i'm even more confused, because here you _are_ always setting `PAGE_DESCRIPTOR` but in the above function you are always clearing it.

Reply:
see previous reply